### PR TITLE
Wip/hughsie/report per remote

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

**Comment by superm1** [comment]
It's mostly good, and I think the test for checking the boot is a great idea.
I did a test run with the fail, and it did report the failure succesfully.  I then resolved the failure (fixed the fwupdate problem) it reported success like it should.  However in the metadata the "generic" error is still there even though it was deemed and probably needs to be removed.

> 
> { 
> "Checksum": "c92f51d5a9558540592568368f079436a3fa2ef5", 
> "Created": 1515769248, 
> "DeviceId": "79f2996ff11608674c5d2e8db20e10e0b646f2cb", 
> "Flags": 299, "Guid": "33773727-8ee7-4d81-9fa0-57e8d889e1fa", 
> "Metadata": 
>{
> "AppstreamGlibVersion": "0.7.1", 
> "BootTime": "1515769219", 
> "CpuArchitecture": "x86_64", 
> "DistroId": "ubuntu", 
> "DistroVersion": "17.10", 
> "FwupdVersion": "1.0.4", 
> "GUsbVersion": "0.2.11", 
> "KernelVersion": "4.14.0+" 
>}, 
> "Modified": 1515769489, 
> "Plugin": "uefi", 
> **"UpdateError": "failed to run update on reboot",** 
> "UpdateState": 2, 
> "VersionNew": "0.1.6.1", 
> "VersionOld": "0.1.5.1"
>  }

File: src/fu-engine.c [hunk 1]
```diff
@@ -3319,26 +3350,33 @@ fu_engine_update_history_device (FuEngine *self, FuDevice *dev_history, GError *
 		return TRUE;
 	}
 
-	/* find the plugin that started the update */
+	/* does the plugin knows the update failure */
```
Comment (superm1) [inline src/fu-engine.c]: /s/knows/know

File: src/fu-engine.c [hunk 2]
```diff
@@ -3319,26 +3350,33 @@ fu_engine_update_history_device (FuEngine *self, FuDevice *dev_history, GError *
 		return TRUE;
```
Comment (superm1) [inline src/fu-engine.c]: Probably need to explicitly clear the error with `fu_device_set_update_error `here.

**Comment by hughsie** [comment]
We could do something like:

	if (update_state == FWUPD_UPDATE_STATE_SUCCESS) {
		if (!fu_history_set_error_msg (self, device, NULL, error))
			return FALSE;
	}

in `fu_history_set_update_state` -- i.e. I can't ever see of a case where the state is SUCCESS and the error is non-NULL.

**Comment by superm1** [comment]
Yeah that's fine.

**Comment by hughsie** [comment]
Split up and fixed up, thanks.

**Pull request closed without merging**